File-based JWT authentication

charlie-gallagher opened the pull request:
Hey @benrwoodard I took a couple minutes to convert JWT authentication from the crude environment variable approach (all my fault...) to this much nicer file-based approach. It functions the same as with `googleanalyticsr`, like we discussed. 

IMPORTANT: I wasn't the one who downloaded/generated our json file containing our authentication info -- I just inherited it. Do you know if it's possible to download the credentials, or do we need to clarify how to format the JSON file as well? 

- The environment variable `AW_AUTH_FILE` contains the file path to the authentication JSON
- You can also pass any path with the `file` argument to `aw_auth`. 
- I updated the `aw_auth` documentation as well. I am on my way to updating the readme documentation.

Example: 

```r
devtools::load_all(".")

aw_auth("jwt", file = "path/to/authfile.json")
```

Checklist: 

- [x] R CMD check ran with 0 errors, 0 warnings, and 0 notes
- [x] I updated documentation with `devtools::document()`

charlie-gallagher commented:
For what it's worth, I include a description of the JSON file fields in `?aw_auth`

benrwoodard commented on man/aw_auth.Rd:
```diff
@@ -8,15 +8,7 @@
-  jwt_token = NULL,
-  ...
-)
+auth_jwt(file = Sys.getenv("AW_AUTH_FILE"), jwt_token = NULL, ...)
```
I love this a lot minus the private key file having to be included.

benrwoodard commented:
Regarding the json file format that is downloaded from inside the developer.adobe console: The json file is downloaded and looks like this --> 
<img width="602" alt="image" src="https://user-images.githubusercontent.com/2105463/153736477-c6286585-a378-4fc9-a6a4-6e557c06c998.png"> 

Preferably we would not require the end-user to edit this file in any way and simply make another argument that would identify where the private.key file is located.

charlie-gallagher commented:
Perfect, thank you! I'll use this as the reference

charlie-gallagher commented:
@benrwoodard is API_KEY the same as CLIENT_ID? I don't see CLIENT_ID in the json

benrwoodard commented:
Yes

charlie-gallagher commented:
I changed things around slightly: 

- JWT authentication requires only two environment variables: `AW_AUTH_FILE` and `AW_PRIVATE_KEY`
- I updated the names to match the file you sent me. I also documented these in `?aw_auth`
- I added a deprecation warning if you try to auth with JWT in the old way
  - Maybe we want to continue supporting the old method? Wouldn't be a huge lift I think.

benrwoodard commented:
@charlie-gallagher One quick question. I have a situation right now where the client will not give me the JSON file but has given me all the variables needed.  Have you already described a way of manually creating the JSON file for auth?
